src/main/java/seedu/address/logic/commands/ReminderDeleteCommand.java

```diff
@@ -0,0 +1,71 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.reminder.Reminder;
+
+/**
+ *  Deletes a reminder in the reminder list.
+ */
+public class ReminderDeleteCommand extends ReminderCommandGroup {
+
+    public static final String COMMAND_SPECIFIER = "delete";
+    public static final String COMMAND_WORD = COMMAND_GROUP + " " + COMMAND_SPECIFIER;
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Deletes a reminder from the reminder list using its index"
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " " + COMMAND_SPECIFIER + " 1 ";
+
+    public static final String MESSAGE_SUCCESS = "Reminder deleted: %1$s";
+
+    private final Index index;
+
+    /**
+     * @param index of the reminder in the reminder list to delete
+     */
+    public ReminderDeleteCommand(Index index) {
+        requireNonNull(index);
+        this.index = index;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        List<Reminder> reminderList = model.getCurrentReminderList();
+
+        if (index.getZeroBased() >= reminderList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_REMINDER_DISPLAYED_INDEX);
+        }
+
+
+        Reminder toDelete = reminderList.get(index.getZeroBased());
+        model.deleteReminder(toDelete);
```

Comment:
is it possible to use optional.orElseThrow here?

Reply:
The index here is not stored in an optional.